### PR TITLE
Map component bug fix 

### PR DESCRIPTION
**Pull request by robdiorio34**
###Problem
Map was white screening when list of multiple addresses had at least one invalid address
###Solution
Filter out invalid address based on null longitude and latitude received from geocoder so only valid addresses are displayed 
###Notes
Refactored mobile and web code, bug fix for undefined/invalid addresses, now filtered out of displayed map markers

### CODE REVIEW

**Comment by robdiorio34** [comment]
[Map component doesn't load if one address is wrong](https://trello.com/c/TVJHszln/61-map-component-doesnt-load-if-one-address-is-wrong)

File: src/Map/index.js [hunk 1]
```diff
@@ -122,7 +122,9 @@ export default class Map extends Component {
           ? [markerAddress]
           : []
         : markerCollection
-        ? markerCollection.map((m) => m.markers_list.markerAddress)
+        ? markerCollection.map((m) => {
```
Comment (scott-massey) [inline src/Map/index.js:125]: Let's revert this change, it's unnecessary

File: src/Map/config.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export const markerWidth = 25
 export const markerHeight = 41
-export const geocodeURL = "https://adalo-geocoder.herokuapp.com/"
+export const geocodeURL = 'https://873fcb166459.ngrok.io/'
```
Comment (scott-massey) [inline src/Map/config.js:3]: We'll need to change this URL before you publish

File: src/Map/map.js [hunk 3]
```diff
@@ -41,11 +41,49 @@ export const getMap = (
   const LATITUDE_DELTA = Math.exp(Math.log(360) - (zoom + 1) * Math.LN2)
   const LONGITUDE_DELTA = LATITUDE_DELTA * (width / height)
 
+  let filteredMarkers = []
+
+  if (isSimple) {
+    filteredMarkers.push({
+      //might need coordinates : longitude:
```
Comment (scott-massey) [inline src/Map/map.js:48]:
```suggestion
```

File: src/Map/map.js [hunk 4]
```diff
@@ -41,11 +41,49 @@ export const getMap = (
...
+      lat: addresses.length > 0 ? addresses[0].location.lat : null,
```
Comment (scott-massey) [inline src/Map/map.js:49]:
```suggestion
      lat: addresses.length > 0 && addresses[0] ? addresses[0].location.lat : null,
```

File: src/Map/map.js [hunk 5]
```diff
@@ -41,11 +41,49 @@ export const getMap = (
...
+      lng: addresses.length > 0 ? addresses[0].location.lng : null,
```
Comment (scott-massey) [inline src/Map/map.js:50]:
```suggestion
      lng: addresses.length > 0 && addresses[0] ? addresses[0].location.lng : null,
```

File: src/Map/map.js [hunk 6]
```diff
@@ -41,11 +41,49 @@ export const getMap = (
...
+      style: { alignItems: 'center', justifyContent: 'center' },
```
Comment (scott-massey) [inline src/Map/map.js:51]:
```suggestion
```
Let's just put this on the component instead of in the array

File: src/Map/map.js [hunk 7]
```diff
@@ -41,11 +41,49 @@ export const getMap = (
...
+      return {
+        lat:
+          addresses.length > 0 && addresses[index]
+            ? addresses[index].location.lat
+            : null,
+        lng:
+          addresses.length > 0 && addresses[index]
+            ? addresses[index].location.lng
+            : null,
+        style: { alignItems: 'center', justifyContent: 'center' },
```
Comment (scott-massey) [inline src/Map/map.js:67]:
```suggestion
```

File: src/Map/map.js [hunk 8]
```diff
@@ -41,11 +41,49 @@ export const getMap = (
...
+      }
+    })
+  }
+
+  filteredMarkers = filteredMarkers.filter((marker) => marker.lat)
+
   const viewCenter =
-    addresses.length > 0
-      ? { lat: addresses[0].location.lat, lng: addresses[0].location.lng }
+    filteredMarkers.length > 0
```
Comment (scott-massey) [inline src/Map/map.js:82]:
```suggestion
    filteredMarkers.length > 0 && filteredMarkers[0]
```

File: src/Map/map.js (hunk 6 above)
Comment (scott-massey) [reply src/Map/map.js:51]: It doesn't actually look like this gets used

**Review by scott-massey — COMMENTED** [review]
some comments...

File: src/Map/index.js (hunk 1 above)
Comment (scott-massey) [reply src/Map/index.js:140]: bump this

File: src/Map/index.js [hunk 9]
```diff
@@ -160,17 +163,57 @@ export default class Map extends Component {
     }
   }
 
+  getFilteredAddresses = () => {
+    const { addresses } = this.state
+
+    const { markerType, onPress, markerCollection, markerImage, markerSource } =
+      this.props
+
+    console.log('Markers', markerCollection)
```
Comment (scott-massey) [inline src/Map/index.js:172]:
```suggestion
```

File: src/Map/index.js [hunk 10]
```diff
@@ -179,6 +222,10 @@ export default class Map extends Component {
       loaded,
     } = this.state
 
+    const filteredMarkers = this.getFilteredAddresses()
+
+    console.log('Filtered: ', filteredMarkers)
```
Comment (scott-massey) [inline src/Map/index.js:227]:
```suggestion
```

File: src/Map/map.js (hunk 8 above)
Comment (scott-massey) [reply src/Map/map.js:37]: bump

File: src/Map/map.web.js [hunk 11]
```diff
@@ -15,27 +15,22 @@ export const addNativeEvent = (apiKey) => {
...
+}) => {
   const defaultCenter = {
     lat: 40.7831,
     lng: -73.9712,
   }
+
   const viewCenter =
-    addresses.length > 0
-      ? { lat: addresses[0].location.lat, lng: addresses[0].location.lng }
+    filteredMarkers.length > 0
```
Comment (scott-massey) [inline src/Map/map.web.js:32]: See my comment on the mobile version

File: src/Map/map.web.js [hunk 12]
```diff
@@ -45,66 +40,28 @@ export const getMap = (
...
           const bounds = new google.maps.LatLngBounds()
-          for (let i = 0; i < addresses.length; i++) {
-            const marker = addresses[i]
-            const newPoint = new google.maps.LatLng(
-              marker.location.lat,
-              marker.location.lng
-            )
+          for (let i = 0; i < filteredMarkers.length; i++) {
+            const marker = filteredMarkers[i]
+            console.log('Marker', marker)
```
Comment (scott-massey) [inline src/Map/map.web.js:47]:
```suggestion
```

**Review by scott-massey — COMMENTED** [review]
Just a few comments! Nothing major

**Review by scott-massey — APPROVED** [review]
Awesome work!

**Review by scott-massey — APPROVED** [review]
